test/SlowTests/Issues/RavenDB-19481.cs

```diff
@@ -0,0 +1,220 @@
+                        MinimumRevisionsToKeep = 100
+                    }
+                });
+
+                using (var session = store.OpenAsyncSession(new SessionOptions
+                {
+                    TransactionMode = TransactionMode.ClusterWide
+                }))
+                {
+                    await session.StoreAsync(new User
+                    {
+                        Name = "Grisha"
+                    }, id);
+                    await session.SaveChangesAsync();
+                }
+
+                await WaitAndAssertForValueAsync(async () =>
+                {
+                    using (var session = store.OpenAsyncSession())
+                    {
+                        var u = await session.LoadAsync<User>(id);
+                        return u == null ? $"\"{id}\" is null" : u.Name;
+                    }
+                }, "Grisha");
+
+                var stats = await store.Maintenance.SendAsync(new GetEssentialStatisticsOperation());
+                Assert.Equal(1, stats.CountOfRevisionDocuments);
+
+                var restoredDatabaseName = $"restored_database-{Guid.NewGuid()}";
+                var backupPath = NewDataPath(suffix: "BackupFolder");
+                var config = Backup.CreateBackupConfiguration(backupPath);
+                RestoreBackupConfiguration restoreConfig;
+                if (options.DatabaseMode == RavenDatabaseMode.Sharded)
+                {
+                    var waitHandles = await Sharding.Backup.WaitForBackupToComplete(store);
+                    await Sharding.Backup.UpdateConfigurationAndRunBackupAsync(Server, store, config);
+                    Assert.True(WaitHandle.WaitAll(waitHandles, TimeSpan.FromMinutes(1)));
+                    var dirs = Directory.GetDirectories(backupPath);
+                    Assert.Equal(3, dirs.Length);
+                    var sharding = await Sharding.GetShardingConfigurationAsync(store);
+                    var settings = Sharding.Backup.GenerateShardRestoreSettings(dirs, sharding);
+                    restoreConfig = new RestoreBackupConfiguration { DatabaseName = restoredDatabaseName, ShardRestoreSettings = settings };
+                }
+                else
+                {
+                    var waitHandles = await Backup.WaitForBackupToComplete(store);
+                    await Backup.UpdateConfigAndRunBackupAsync(Server, config, store);
+                    restoreConfig = new RestoreBackupConfiguration { BackupLocation = Directory.GetDirectories(backupPath).First(), DatabaseName = restoredDatabaseName };
+                    Assert.True(WaitHandle.WaitAll(waitHandles, TimeSpan.FromMinutes(1)));
+                }
+
+                var clusterTransactions = new Dictionary<string, long>();
+                Server.ServerStore.ForTestingPurposesOnly().BeforeExecuteClusterTransactionBatch = (dbName, batch) =>
+                {
+                    if (dbName == restoredDatabaseName)
```

Comment:
I wonder if this is working for sharded... if you restore to a sharded database the name is `foo$1`

Reply:
changed to `StartWith`